### PR TITLE
Initial cleanup

### PR DESCRIPTION
**Pull request by mrlukasbos**
Remove a lot of unused files/classes and cleans up the code by structuring into components and functions, and by adding documentation.

needs https://github.com/RoboTeamTwente/roboteam_world/pull/36

### CODE REVIEW

File: include/roboteam_world/kalman/KalmanFilter.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  KalmanFilter();
+  void kalmanUpdate();
+  void newFrame(const proto::SSL_DetectionFrame &msg);
+
+  proto::World getWorld();
+  KalmanRobot blueBots[BOTCOUNT];
+  KalmanRobot yellowBots[BOTCOUNT];
+  KalmanBall ball;
+
+  std::mutex filterMutex;
```
Comment (rolfvdhulst) [inline include/roboteam_world/kalman/KalmanFilter.h]: Is this used anywhere? As far as I know there is no multithreading in world.

File: include/roboteam_world/predictor.h [hunk 2]
```diff
@@ -1,46 +0,0 @@
-#pragma once
```
Comment (rolfvdhulst) [inline include/roboteam_world/predictor.h]: This file is as far as I know the last file that uses boost (maybe some utils class still uses it?)

Comment (rolfvdhulst) [reply include/roboteam_world/predictor.h]: So we can probably remove it for good.

File: src/WorldHandler.cpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+void WorldHandler::handleRefboxPackets(proto::SSL_Referee &ref_packet) const {
+    while (refbox_client && refbox_client->receive(ref_packet)) {
+        ref_pub->send(ref_packet);
+    }
+}
+
+void WorldHandler::handleVisionPackets(proto::SSL_WrapperPacket &vision_packet) const {
+    while (vision_client && vision_client->receive(vision_packet)) {
+        if (vision_packet.has_detection()) KF->newFrame(vision_packet.detection());
+        if (vision_packet.has_geometry()) geom_pub->send(vision_packet.geometry());
```
Comment (rolfvdhulst) [inline src/WorldHandler.cpp]: I would like if these if statements were scoped so you cannot make mistakes when editing the code in the future and for readability.

File: src/kalman/KalmanFilter.cpp [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+// Created by kjhertenberg on 13-5-19.
+//
+
+#include "roboteam_proto/messages_robocup_ssl_detection.pb.h"
+#include <kalman/KalmanFilter.h>
+
+namespace world {
+
+KalmanFilter::KalmanFilter() {
+  std::lock_guard<std::mutex> lock(filterMutex);
```
Comment (rolfvdhulst) [inline src/kalman/KalmanFilter.cpp]: The mutex should not be necessary since WorldHandler is single threaded.

Comment (rolfvdhulst) [reply src/kalman/KalmanFilter.cpp]: (Goes throughout the file)

File: test/KalmanTest.cpp [hunk 5]
```diff
@@ -87,55 +76,55 @@ namespace rtt {
...
-            frametest.kalmanUpdate();
-        }
-
-        Position testX = frametest.getPos(0);
-        EXPECT_FLOAT_EQ(testX.x, 2);
-        EXPECT_FLOAT_EQ(testX.y, 2);
-        EXPECT_FLOAT_EQ(testX.rot, 0);
-    }
+//
+//    TEST(KalmanTest, ZXK) {
```
Comment (rolfvdhulst) [inline test/KalmanTest.cpp]: Were these tests failing? In any case I don't like to see commented code.

**Review by rolfvdhulst — COMMENTED** [review]
Looks good, I have no major comments. I'll approve once someone has tested this and the branch compiles properly.

File: include/roboteam_world/kalman/KalmanFilter.h (hunk 1 above)
Comment (rolfvdhulst) [reply include/roboteam_world/kalman/KalmanFilter.h]: Not removing it since it only adds safety.

**Review by rolfvdhulst — APPROVED** [review]

File: src/WorldHandler.cpp [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    roboteam_utils::Timer t;
+    t.loop([&]() {
+      handleVisionPackets(vision_packet);
+      handleRefboxPackets(ref_packet);
+
+      KF->kalmanUpdate();
+      world_pub->send(KF->getWorld());
+    }, 100);
+}
+
```
Comment (RobotJesse) [inline src/WorldHandler.cpp]: I really like this class!

**Review by RobotJente — COMMENTED** [review]
These changes make the class very readable! I will test the latest changes with grSim, if there is no change in behaviour I think we should merge them. Will also test with switching sides to see if the ourbots -> yellowbots refactor gives any problems.

**Comment by RobotJesse** [comment]
Nothing breaks in grSim, so let's merge this

**Review by RobotJente — APPROVED** [review]
Let's merge this if @HaicoDorenbos agrees

File: test/KalmanTest.cpp (hunk 5 above)
Comment (HaicoDorenbos) [reply test/KalmanTest.cpp]: Fully agree, because we are using Git there is no need to have commented code.

File: src/kalman/KalmanRobot.cpp [hunk 7]
```diff
@@ -0,0 +1,40 @@
+#include "kalman/KalmanRobot.h"
+
+namespace world {
+
+KalmanRobot::KalmanRobot() : KalmanRobot(99) {
```
Comment (HaicoDorenbos) [inline src/kalman/KalmanRobot.cpp]: Good usage of inheritance :+1:

**Review by HaicoDorenbos — CHANGES_REQUESTED** [review]
I will approve this pull request if the commented code in test/KalmanTest.cpp is removed (might also be done by Jesse or me). Commented code is not needed if you use Git and can be dangerous for the development/maintenance of the code as explained in: https://kentcdodds.com/blog/please-dont-commit-commented-out-code

Moreover the new team has informally decided to document every class using Javadoc (with a short summary, author and since) and every public function (with a short summary, param and return). But since these rules only apply for the new team, it is not required to change this.

**Review by HaicoDorenbos — APPROVED** [review]
Commented test cases are put at the fix/KalmanTests branch and a new issue is created for this problem.

**Pull request merged**